src/make_watertight/Gen.cpp

```diff
@@ -0,0 +1,2117 @@
+// the correct vert is not the closest. For example, iter_surf4010 the skin
+// loop has the same point in it twice, at two different locations (center of L).
+// This ensure that both are returned as candidates.
+moab::ErrorCode Gen::find_closest_vert(const double tol,
+                                       const moab::EntityHandle reference_vert,
+                                       const std::vector<moab::EntityHandle> loop_of_verts,
+                                       std::vector<unsigned>& positions,
+                                       std::vector<double>& dists) {
+
+  moab::ErrorCode rval;
+  positions.clear();
+  dists.clear();
+  const double TOL_SQR = tol * tol;
+  unsigned min_pos;
+  double sqr_min_dist = std::numeric_limits<double>::max();
+  for (unsigned int i = 0; i < loop_of_verts.size(); i++) {
+    double sqr_dist = std::numeric_limits<double>::max();
+    rval = squared_dist_between_verts(reference_vert, loop_of_verts[i], sqr_dist);
+    MB_CHK_SET_ERR(rval, "could not get dist");
+    if (sqr_dist < sqr_min_dist) {
+      if (sqr_dist >= TOL_SQR) {
+        sqr_min_dist = sqr_dist;
+        min_pos = i;
+      } else {
+        sqr_min_dist = TOL_SQR;
+        positions.push_back(i);
+        dists.push_back(sqrt(sqr_dist));
+      }
+    }
+  }
+
+  if (dists.empty()) {
+    dists.push_back(sqrt(sqr_min_dist));
+    positions.push_back(min_pos);
+  }
+
+
+  return moab::MB_SUCCESS;
+}
+
+moab::ErrorCode Gen::merge_vertices(moab::Range verts /* in */, const double tol /* in */) {
+
+  moab::ErrorCode result;
+  const double SQR_TOL = tol * tol;
+  // Clean up the created tree, and track verts so that if merged away they are
+  // removed from the tree.
+  moab::AdaptiveKDTree kdtree(MBI()); //, true, 0, moab::MESHSET_TRACK_OWNER);
+  // initialize the KD Tree
+  moab::EntityHandle root;
+  const char settings[] = "MAX_PER_LEAF=6;MAX_DEPTH=50;SPLITS_PER_DIR=1;PLANE_SET=2;MESHSET_FLAGS=0x1;TAG_NAME=0";
+  moab::FileOptions fileopts(settings);
+  // builds the KD Tree, making the moab::EntityHandle root the root of the tree
+  result = kdtree.build_tree(verts, &root, &fileopts);
+  assert(moab::MB_SUCCESS == result);
+  // create tree iterator to loop over all verts in the tree
+  moab::AdaptiveKDTreeIter tree_iter;
+  kdtree.get_tree_iterator(root, tree_iter);
+  for (moab::Range::iterator i = verts.begin(); i != verts.end(); ++i) {
+    double from_point[3];
+    result = MBI()->get_coords(&(*i), 1, from_point);
+    assert(moab::MB_SUCCESS == result);
+    std::vector<moab::EntityHandle> leaves_out;
+    result = kdtree.distance_search(from_point, tol, leaves_out, root);
+    assert(moab::MB_SUCCESS == result);
+    for (unsigned int j = 0; j < leaves_out.size(); j++) {
+      std::vector<moab::EntityHandle> leaf_verts;
+      result = MBI()->get_entities_by_type(leaves_out[j], moab::MBVERTEX, leaf_verts);
+      assert(moab::MB_SUCCESS == result);
+      if (100 < leaf_verts.size())
```

Comment:
Readability: I would leave a comment here explaining the significance of `100`, and why it merits additional logging for this case.